src/order/preorder_hom.lean

```diff
@@ -65,6 +65,14 @@ by { ext, refl }
 def subtype.val (p : α → Prop) : subtype p →ₘ α :=
 ⟨subtype.val, λ x y h, h⟩
 
+-- TODO[gh-6025]: make this a global instance once safe to do so
+local attribute [instance]
+def unique [subsingleton α] : unique (α →ₘ α) :=
+{ default := preorder_hom.id, uniq := λ a, ext _ _ (subsingleton.elim _ _) }
+
+lemma preorder_hom_eq_id [subsingleton α] (g : α →ₘ α) : g = preorder_hom.id :=
```

Comment:
```suggestion
@[simp] lemma preorder_hom_eq_id [subsingleton α] (g : α →ₘ α) : g = preorder_hom.id :=
```
Whoops

Reply:
Nope, I did it. The linter complained about it.